app/src/main/java/com/mapbox/navigation/examples/standalone/background/BackgroundExampleActivity.kt

```diff
@@ -0,0 +1,474 @@
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.core.lifecycle.requireMapboxNavigation
+import com.mapbox.navigation.core.replay.route.ReplayRouteSession
+import com.mapbox.navigation.core.trip.session.LocationMatcherResult
+import com.mapbox.navigation.core.trip.session.LocationObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.examples.R
+import com.mapbox.navigation.examples.databinding.MapboxActivityBackgroundExampleBinding
+import com.mapbox.navigation.ui.maps.NavigationStyles
+import com.mapbox.navigation.ui.maps.camera.NavigationCamera
+import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource
+import com.mapbox.navigation.ui.maps.camera.lifecycle.NavigationBasicGesturesHandler
+import com.mapbox.navigation.ui.maps.camera.transition.NavigationCameraTransitionOptions
+import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
+import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
+import com.mapbox.navigation.ui.voice.api.MapboxAudioGuidance
+
+/**
+ * This example demonstrates how to keep navigation session running while activity is destroyed.
+ * The active guidance session stops only when user explicitly stop is by pressing the "stop" button.
+ *
+ * Before running the example make sure you have put your access_token in the correct place
+ * inside [app/src/main/res/values/mapbox_access_token.xml]. If not present then add this file
+ * at the location mentioned above and add the following content to it
+ *
+ * <?xml version="1.0" encoding="utf-8"?>
+ * <resources xmlns:tools="http://schemas.android.com/tools">
+ *     <string name="mapbox_access_token"><PUT_YOUR_ACCESS_TOKEN_HERE></string>
+ * </resources>
+ *
+ * The example assumes that you have granted location and notification permissions and does not
+ * enforce it. However,the permission is essential for proper functioning of this example.
+ * The example also uses replay location engine to facilitate navigation
+ * without actually physically moving.
+ *
+ * How to use this example:
+ * - You can long-click the map to select a destination.
+ * - The guidance will start to the selected destination while simulating location updates.
+ * - You can leave activity but active guidance session will be running.
+ * - Press the "Stop" button to finish the session. If you leave activity during free drive session,
+ * MapboxNavigation will be destroyed.
+ */
+class BackgroundExampleActivity : AppCompatActivity() {
```

Comment:
There's a lot of boiler-plate code here. It makes it harder to track which parts are important for background. Can we remove all the unnecessary code from here and just keep the bare minimum?